Modules/Quotas/src/Quotas.Domain/Aggregates/Tiers/Tier.cs

```diff
@@ -44,6 +46,31 @@ public Tier(TierId id, string name)
         return Result.Success<TierQuotaDefinitionId, DomainError>(quotaDefinition.Id);
     }
 
+    public IEnumerable<Result<TierQuotaDefinition, DomainError>> CreateQuotaForAllMetrics(IEnumerable<Metric> metrics)
```

Comment:
The method name should probably make it clear that it can only be called for the QueuedForDeletion tier.